### PR TITLE
PS-3968: Tune clang-format diff checkings for existing SQL and InnoDB style

### PR DESCRIPTION
**Pull request by inikep**
Create two `.clang-format` config files: one for InnoDB and one for the rest (SQL).
The files are compatible with clang-format-5.0 or newer.

### CODE REVIEW

**Review by laurynas-biveinis — COMMENTED** [review]
How the styles were verified to match the SQL/InnoDB styles?

**Comment by inikep** [comment]
I tried manually to minimize differences between original files and clang-formatted. There is no config that will 100% match all files as I found different formatting styles even in the same file. This PR will only reduce a number of formatting errors from Travis. Moreover adding `-style=file` causes that a proper submodule `.clang-format` for RocksDB and TokuDB will be used.

**Review by laurynas-biveinis — APPROVED** [review]

File: storage/innobase/.clang-format [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+SpacesInAngles:  false
+SpacesInContainerLiterals: true
+SpacesInCStyleCastParentheses: false
+SpacesInParentheses: false
+SpacesInSquareBrackets: false
+Standard:        Auto
+TabWidth:        8
+
+
+# changes for MySQL 5.x
```
Comment (percona-ysorokin) [inline storage/innobase/.clang-format]: `changes for MySQL 5.x (InnoDB)`

**Review by percona-ysorokin — APPROVED** [review]
LGTM
minor comment can be ignored.

**Pull request merged**